Remove discontinued and unmaintained boards

nashif opened the pull request:
Removing support for galileo, arduino_101 and Quark SE C1000 and their variants.

nashif committed:
boards: remove galileo board

This board and SoC was discontinued some time ago and is currently not
maintained in the zephyr tree.
Remove all associated configurations and variants from the tree.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: remove arduino 101 and related boards

This board and SoC was discontinued some time ago and is currently not
maintained in the zephyr tree.
Remove all associated configurations and variants from the tree.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: remove quarl_se_c1000

This board and SoC was discontinued some time ago and is currently not
maintained in the zephyr tree.
Remove all associated configurations and variants from the tree.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
doc: minnowboard: add grub docs

Was previously documented as part of the galileo board. Moving here now.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: remove code for interrupt forwarding bug

This only applied to quark_se, so removing it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

pfalcon reviewed (commented):
arduino_101 is an important sample board for embedded x86 functionality, it's the default test board for some Zephyr-related projects, e.g. https://github.com/intel/zephyr.js .

I myself have a galileo lying somewhere around.

I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?). 

Summing up, why remove well-known, popular (large amounts still in the community) and useful boards?

carlescufi reviewed (approved):
LGTM

galak commented:
Shouldn't we be marking these boards as BOARD_DEPRECATED first?

andrewboie commented on tests/subsys/fs/fat_fs_api/README.txt:
```diff
@@ -10,7 +10,7 @@ Building and Running Project:
 The demo will run on Arduino 101 and will use the on-board SPI flash.
```
still mentions Arduino 101

andrewboie commented on scripts/sanity_chk/sanity_compare.args:
```diff
@@ -8,11 +8,3 @@ frdm_k64f
--p
-quark_se_c1000_ss_devboard
--p
-altera_max10
```
why was altera_max10 removed? It's a Nios II board

nashif replied:
altera_max10 is not being removed, here just cleaning up this more than I should have I guess.

gnuless commented:
> arduino_101 is an important sample board for embedded x86 functionality,

It's been discontinued for two years.

> I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?).

I'm not sure why the Minnowboard is being yanked. @nashif Minnowboard isn't a Quark, do we really want to drop support?

gnuless commented:
> I believe there were commits for minnowboard until very recently (was it target of some hypervisor work or something?).

Sorry, @nashif, I think @pfalcon is mistaken and I didn't look closely, the Minnowboard isn't being removed here.

nashif committed:
toolchain: remove issm support

This toolchain is no londer supported or needed. It was used to build
configurations that are now being removed.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

tbursztyka reviewed (approved):
I am gladly approving that PR!

Goodbye awful galileo board...


oh and quark_se ones too, yes no more qmsi

dbkinder reviewed (approved):
There's material out there about running Zephyr on some of these boards.  But I suppose we can aim them at the LTS (1.14) release instead of master.
